src/nameof.ts

```diff
@@ -32,17 +32,11 @@ export function nameof<T extends Object>(nameFunction: ((obj: T) => any) | { new
 
     // ES5 prop selector:
     // "function (x) { return x.prop; }"
-    if ((new RegExp(/function(\s)?\(/, "g")).test(fnStr)) {
-        const firstDotIndex = fnStr.indexOf(".");
-        const semicolonIndex = fnStr.indexOf(";");
-
-        return cleanseAssertionOperators(
-            fnStr.substring(
-                firstDotIndex + 1,
-                semicolonIndex
-            )
-        );
-    }
+    // webpack production build excludes the spaces and optional trailing semicolon:
+    //   "function(x){return x.prop}"
+    // FYI - during local dev testing i observed carriage returns after the curly brackets as well
+    var ES5match = fnStr.match(/function\s*\(\w\)\s*\{[\r\n\s]*return\s*\w+\.(\w+)/i);
```

Comment:
There are a few problems with this regex.

1. The first `\(\w\)` (after `function\s*`) should be `\(\w+\)` (with a `+`) instead because otherwise `x => x.prop` will match, but `xy => xy.prop` will not. The `return` portion of the regex does account for `\w+`, so the parameter portion should match.

2. This regex does not account for multiple properties after the first `.` in the return statement. `x => x.prop` does return `prop`, but `x => x.prop.child.grandchild` only returns `prop` rather than `prop.child.grandchild` as it should.

3. I believe `return\s*` should instead be `return\s+` because we definitely want to capture the portion followed by `return ` (including the space(s) after `return`).

Additionally, tslint should have already warned about this, but `const` should be used in place of `var` as the variable is not reassigned.

Also, you may have already done so, but make sure you run the tests for the lib: `npm run test:es5` and `npm run test:es6`.

Additional stylistic suggestions: I would prefer `ES5match` to be camel cased, i.e. `es5Match` and one new line between the variable's declaration and the if block below.